File: backend/mood_logic.py
import json
import os

CONFIG_PATH = os.path.join(os.path.dirname(__file__), 'config.json')
with open(CONFIG_PATH, 'r') as file:
    config = json.load(file)

mood_keywords = config["mood_keywords"]
response_keywords = config["response_keywords"]
recommendations = config["recommendations"]

def detect_mood(message):
    message = message.lower()
    for mood, keywords in mood_keywords.items():
        if any(keywords in message for keywords in keywords):
            return mood
    return "fallback_soft"

def detect_response_intent(message):
    message = message.lower()
    for intent, keywords in response_keywords.items():
        if any(keywords in message for keywords in keywords):
            return intent
    return "fallback_soft"

def get_recommendation(mood):
    return recommendations.get(mood, {})

def intro_message():
    return (
        "Hey! I'm YARA, your personal movie planner\n"
        "Tell me your mood and I’ll find the perfect movie — and if you're up for it, "
        "I’ll suggest snacks, drinks, a theme, and even games.\n"
        "So… how are you feeling?"
    )

File: backend/mood_logic.js

const fs = require("fs");
const path = require("path");

const configPath = path.join(__dirname, "config.json");
const config = JSON.parse(fs.readFileSync(configPath, "utf8"));

const moodKeywords = config.mood_keywords;
const responseKeywords = config.response_keywords;
const recommendations = config.recommendations;

function detectMood(message) {
  const msg = message.toLowerCase();
  for (const mood in moodKeywords) {
    if (moodKeywords[mood].some((kw) => msg.includes(kw))) {
      return mood;
    }
  }
  return "fallback_soft";
}

function detectResponseIntent(message) {
  const msg = message.toLowerCase();
  for (const intent in responseKeywords) {
    if (responseKeywords[intent].some((kw) => msg.includes(kw))) {
      return intent;
    }
  }
  return "fallback_soft";
}

function getRecommendation(mood) {
  return recommendations[mood] || {};
}

function introMessage() {
  return (
    "Hey! I'm YARA, your personal movie planner\n" +
    "Tell me your mood and I’ll find the perfect movie — and if you're up for it, " +
    "I’ll suggest snacks, drinks, a theme, and even games.\n" +
    "So… how are you feeling?"
  );
}

module.exports = {
  detectMood,
  detectResponseIntent,
  getRecommendation,
  introMessage
};
